src/utils/merge-deep.ts

```diff
@@ -164,6 +164,10 @@ export default function mergeDeep(
     return source;
   } else if (sourceIsArray) {
     return source;
+  } else if (target === null || target === undefined) {
+    // typeof null === 'object'
```

Comment:
Could you explain the comment in the PR desc and here?  It seems like you ran across a situation where the value was null but we were doing something wrong b/c we let is pass b/c we thought it was `typeof object`

Reply:
yup!

Reply:
done